docs/source/session.rst

```diff
@@ -0,0 +1,138 @@
+Sparkly Session
+===============
+
+``SparklySession`` is the main entry point to sparkly's functionality.
+It's derived from ``SparkSession`` to provide additional features on top of the default session.
+The are two main differences between ``SparkSession`` and ``SparklySession``:
+
+    1. ``SparklySession`` doesn't have ``builder`` attribute,
+       because we prefer declarative style over imperative.
+    2. Hive support is enabled by default.
+
+The next snippet shows how declarative style is used:
+
+.. code-block:: python
+
+    # PySpark-style (imperative)
+    from pyspark import SparkSession
+
+    spark = SparkSession.builder\
+        .appName('My App')\
+        .master('spark://')\
+        .config('spark.sql.shuffle.partitions', 10)\
+        .getOrCreate()
+
+    # Sparkly-style (declarative)
+    from sparkly import SparklySession
+
+    class MySession(SparklySession):
+        options = {
+            'spark.app.name': 'My App',
```

Comment:
Meh, in practice you don't need to touch them. `master` should be configured by spark environment, not in your code.